File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/model/base/personalaccount/AccountType.java
package org.d3javu.backend.model.base.personalaccount;

public enum AccountType {
    PERSONAL("Личный счёт"),
    BUSINESS("Расчетный счет");

    public final String name;

    private AccountType(String name) {
        this.name = name;
    }

}

File: Apps/BankBase/backend/src/main/scala/model/personalaccount/AccountType.scala
package model.personalaccount

import zio.json.{DeriveJsonCodec, JsonCodec}

enum AccountType(defaultName: String) {
  case PERSONAL extends AccountType("Личный счет")
}

object AccountType {
  inline given JsonCodec[AccountType] = DeriveJsonCodec.gen
}